File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    flag = True
    
    heapq.heapify(scoville)
    
    while len(scoville)>0:
        if scoville[0]>=K:
            flag = False
            break
            
        if len(scoville)==1:
            break
            
        a, b = heapq.heappop(scoville), heapq.heappop(scoville)
        heapq.heappush(scoville, a+b*2)
        answer+=1
    
    if flag:
        answer = -1
    
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> pq;
    for (auto& p : scoville){
        pq.push(p);
    }
    /*
    while (!pq.empty()){
        cout << pq.top() << endl;
        pq.pop();
    }
    */
    bool fail = false;
    while (true){
        if (pq.top() < K){
            if (pq.size() > 1){
                int a = pq.top();
                pq.pop();
                int b = pq.top();
                pq.pop();
                pq.push(a+(b*2));
                answer++;
            }
            else{
                fail = true;
                break;
            }
        }
        else{
            break;
        }
    }
    
    if (fail) answer = -1;
    
    
    return answer;
}